cmd/erasure-createfile.go

```diff
@@ -18,85 +18,93 @@ package cmd
 
 import (
 	"context"
-	"hash"
 	"io"
 
+	"sync"
+
 	"github.com/minio/minio/cmd/logger"
 )
 
-// CreateFile creates a new bitrot encoded file spread over all available disks. CreateFile will create
-// the file at the given volume and path. It will read from src until an io.EOF occurs. The given algorithm will
-// be used to protect the erasure encoded file.
-func (s *ErasureStorage) CreateFile(ctx context.Context, src io.Reader, volume, path string, buffer []byte, algorithm BitrotAlgorithm, writeQuorum int) (f ErasureFileInfo, err error) {
-	if !algorithm.Available() {
-		logger.LogIf(ctx, errBitrotHashAlgoInvalid)
-		return f, errBitrotHashAlgoInvalid
+// Writes in parallel to bitrotWriters
+type parallelWriter struct {
+	writers     []*bitrotWriter
+	writeQuorum int
+}
+
+// Append appends data to bitrotWriters in parallel.
+func (p *parallelWriter) Append(ctx context.Context, blocks [][]byte) error {
+	var wg sync.WaitGroup
+	errs := make([]error, len(p.writers))
+	for i, writer := range p.writers {
+		if writer == nil {
+			errs[i] = errDiskNotFound
+			continue
+		}
+
+		wg.Add(1)
+		go func(i int, writer *bitrotWriter) {
+			defer wg.Done()
+			errs[i] = writer.Append(blocks[i])
+			if errs[i] != nil {
+				p.writers[i] = nil
+			}
+		}(i, writer)
 	}
-	f.Checksums = make([][]byte, len(s.disks))
-	hashers := make([]hash.Hash, len(s.disks))
-	for i := range hashers {
-		hashers[i] = algorithm.New()
+	wg.Wait()
+
+	// If nilCount >= p.writeQuorum, we return nil.
```

Comment:
This comment doesn't clarify why we do that. Why do we return nil if `nilCount > p.writeQuorum`?